e2e/mockFetch.ts

```diff
@@ -0,0 +1,28 @@
+import { Hex, sha256 } from 'viem';
+
+export function mockFetch() {
+  const nativeFetch = window.fetch;
+  window.fetch = async function mockedFetch(
+    input: RequestInfo | URL,
+    init?: RequestInit,
+  ) {
+    if (input instanceof Request || !URL.canParse(input))
+      return nativeFetch(input, init);
+
+    const url = new URL(input);
+
+    if (url.hostname === 'swap.p.rainbow.me') {
```

Comment:
intercepts requests to `swap.p.rainbow.me` and responds with our mocked responses